### PR TITLE
More examples

### PR DESCRIPTION
**Pull request by pavanpodila**
Trying with the counter and todos example. Todos is failing right now. @katis need your inputs

### CODE REVIEW

**Comment by pavanpodila** [comment]
It fails with the following error: 

```
error: '_$TodoList.pendingTodos' ('() → ObservableList<dynamic>') isn't a valid override of 'TodoListBase.pendingTodos' ('() → ObservableList<Todo>'). (invalid_override at [flutter_mobx_example] lib/todos.dart:18)
```

Essentially the generic return types are not treated correctly.

**Comment by katis** [comment]
@pavanpodila I'll take a look, my guess is that just getting a type name in codegen doesn't get the generic arguments.

**Comment by pavanpodila** [comment]
Yup, that's the issue. We need a deeper introspection to construct the Type name. Not sure if Element supports that case but wasn't obvious from the API docs.

**Comment by katis** [comment]
Yeah, I think I know how to fix it. Can I push to this branch in a moment?

**Comment by pavanpodila** [comment]
Sure @katis

**Comment by katis** [comment]
@pavanpodila I used element `displayName` instead of `name` and that seems to work.

File: flutter_mobx/example/lib/counter.dart [hunk 1]
```diff
@@ -2,6 +2,21 @@ import 'package:flutter/material.dart';
 import 'package:flutter_mobx/flutter_mobx.dart';
 import 'package:mobx/mobx.dart';
 
+part 'counter.g.dart';
+
+class Counter = CounterBase with _$Counter;
+
+@observable
```
Comment (amrnt) [inline flutter_mobx/example/lib/counter.dart]: You don't need this decoration anymore as the class implements `Store`

**Comment by pavanpodila** [comment]
@katis the `displayName` fix seems to work for `@observable` annotations but still creating computed props without the generic type.

For example:
```dart
  @computed
  ObservableList<Todo> get pendingTodos =>
      todos.where((todo) => todo.done != true);
```
is still creating:

```dart
  @override
  ObservableList get pendingTodos => (_$pendingTodosComputed ??=
          Computed<ObservableList>(() => super.pendingTodos))
      .value;
  Computed<ObservableList> _$completedTodosComputed;
```

Note the loss of the type arguments for `ObservableList`

**Comment by katis** [comment]
Worked for me, just checked out the code and re-ran the codegen.

File: flutter_mobx/pubspec.yaml [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
+description: Flutter Mobx.dart integration.
+version: 0.0.1
+author:
+homepage:
+
+environment:
+  sdk: '>=2.1.0-dev <3.0.0'
+
+dependencies:
+  mobx: ^0.0.19
```
Comment (katis) [inline flutter_mobx/pubspec.yaml]: use the incremented 0.0.20 version

File: flutter_mobx/example/pubspec.yaml [hunk 3]
```diff
@@ -20,10 +20,18 @@ dependencies:
   # The following adds the Cupertino Icons font to your application.
   # Use with the CupertinoIcons class for iOS style icons.
   cupertino_icons: ^0.1.2
+  mobx: ^0.0.19
```
Comment (katis) [inline flutter_mobx/example/pubspec.yaml]: use the incremented 0.0.20 version

Comment (pavanpodila) [reply flutter_mobx/example/pubspec.yaml]: worked after running clean on the `build_runner`

` flutter packages pub run build_runner clean`

**Comment by pavanpodila** [comment]
@katis please review and merge if it looks good

**Pull request merged**